### PR TITLE
postgresql modules: fix sanity issues

### PR DESCRIPTION
**Pull request by Andersson007**
##### SUMMARY
postgresql modules: fix sanity issues

##### ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request

##### COMPONENT NAME
```lib/ansible/modules/database/postgresql/postgresql_copy.py```
...

### CODE REVIEW

**Comment by Andersson007** [comment]
ready_for_review

File: lib/ansible/modules/database/postgresql/postgresql_membership.py [hunk 1]
```diff
@@ -156,8 +156,8 @@
 def main():
     argument_spec = postgres_common_argument_spec()
     argument_spec.update(
-        groups=dict(type='list', aliases=['group', 'source_role', 'source_roles']),
-        target_roles=dict(type='list', aliases=['target_role', 'user', 'users']),
+        groups=dict(type='list', elements='str', required=True, aliases=['group', 'source_role', 'source_roles']),
+        target_roles=dict(type='list', elements='str', required=True, aliases=['target_role', 'user', 'users']),
```
Comment (felixfontein) [inline lib/ansible/modules/database/postgresql/postgresql_membership.py:160]: Is there a valid use-case with `groups` or `target_roles` not specified?

File: lib/ansible/modules/database/postgresql/postgresql_slot.py [hunk 2]
```diff
@@ -222,7 +222,7 @@ def main():
     argument_spec = postgres_common_argument_spec()
     argument_spec.update(
         db=dict(type="str", aliases=["login_db"]),
-        name=dict(type="str", aliases=["slot_name"]),
+        name=dict(type="str", required=True, aliases=["slot_name"]),
```
Comment (felixfontein) [inline lib/ansible/modules/database/postgresql/postgresql_slot.py:225]: Same question here.

File: lib/ansible/modules/database/postgresql/postgresql_subscription.py [hunk 3]
```diff
@@ -612,10 +613,10 @@ def __exec_sql(self, query, check_mode=False):
 def main():
     argument_spec = postgres_common_argument_spec()
     argument_spec.update(
-        name=dict(required=True),
-        db=dict(type='str', aliases=['login_db']),
+        name=dict(type='str', required=True),
+        db=dict(type='str', required=True, aliases=['login_db']),
```
Comment (felixfontein) [inline lib/ansible/modules/database/postgresql/postgresql_subscription.py:617]: and here

File: lib/ansible/modules/database/postgresql/postgresql_tablespace.py [hunk 4]
```diff
@@ -378,7 +379,7 @@ def __set_setting(self, setting):
 def main():
     argument_spec = postgres_common_argument_spec()
     argument_spec.update(
-        tablespace=dict(type='str', aliases=['name']),
+        tablespace=dict(type='str', required=True, aliases=['name']),
```
Comment (felixfontein) [inline lib/ansible/modules/database/postgresql/postgresql_tablespace.py:382]: and here

File: lib/ansible/modules/database/postgresql/postgresql_membership.py (hunk 1 above)
Comment (Andersson007) [reply lib/ansible/modules/database/postgresql/postgresql_membership.py:160]: @felixfontein , no, it isn't. it doesn't make sense, all these parameters are crucial

**Comment by Andersson007** [comment]
ready_for_review

File: lib/ansible/modules/database/postgresql/postgresql_membership.py (hunk 1 above)
Comment (felixfontein) [reply lib/ansible/modules/database/postgresql/postgresql_membership.py:160]: Good, in that case it's fine. I just wanted to make sure that there's no error.

**Comment by felixfontein** [comment]
It might make sense to add a changelog fragment for the modules where the behavior changes (i.e. `required=True` is added), informing the user that error checking has been improved.

**Comment by Andersson007** [comment]
Great suggestion, thanks! It should be there. Will add

**Comment by Andersson007** [comment]
i haven't added a changelog fragment for ```postgresql_subscription``` because it's not been released (coming in 2.10)

**Comment by felixfontein** [comment]
What about `postgresql_tablespace`? That was added for 2.8 and now has a parameter that is `required`.

**Comment by Andersson007** [comment]
Done, thank you for pointing it out! (i slept for 3 hours, better not to code in such a condition:)

**Comment by Andersson007** [comment]
@felixfontein if it is ok now, feel free to merge:)

**Review by felixfontein — APPROVED** [review]

**Comment by Andersson007** [comment]
@felixfontein thanks for reviewing!

**Pull request merged**